From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 13 Nov 2014 20:42:12 +0000
Subject: regulatory: Avoid ABI change in 3.2.64
Forwarded: not-needed

ieee80211_regdomain::alpha2 grew from 2 to 3 bytes, to allow for a
null terminator.  Thankfully the adjacent members both have 32-bit
alignment (on all release architectures) so this only takes 1 of
2 padding bytes and we can hide it from genksyms.

--- a/include/net/regulatory.h
+++ b/include/net/regulatory.h
@@ -92,7 +92,13 @@ struct ieee80211_reg_rule {
 
 struct ieee80211_regdomain {
 	u32 n_reg_rules;
+#ifdef __GENKSYMS__
+	char alpha2[2];
+	/* 2 bytes padding for alignment */
+#else
 	char alpha2[3];
+	/* 1 byte padding for alignment */
+#endif
 	struct ieee80211_reg_rule reg_rules[];
 };
 
